jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -185,6 +188,25 @@ static boolean hasSetOutEncoding() {
 	static Charset getConsoleDefaultCharset() {
 		return consoleDefaultCharset;
 	}
+
+	/*
+	 * Return the appropriate System console using the requested Charset.
+	 */
+	static PrintStream createConsole(FileDescriptor desc, Charset charset) {
+		/*[IF PLATFORM-mz31 | PLATFORM-mz64]*/
+		if (charset == null) {
+			return ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(desc)), true);
+		} else {
+			return ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(desc)), true, charset);
+		}
+		/*[ELSE]*/
+		if (charset == null) {
+			return new PrintStream(new BufferedOutputStream(new FileOutputStream(desc)), true);
+		} else {
+			return new PrintStream(new BufferedOutputStream(new FileOutputStream(desc)), true, charset);
+		}
```

Comment:
Perhaps it's worth extracting the common subexpression `new BufferedOutputStream(new FileOutputStream(desc))`.

Reply:
```suggestion
	return ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(desc)), true, charset == null ?  consoleDefaultCharset : charset);
/*[ELSE]*/
	return new PrintStream(new BufferedOutputStream(new FileOutputStream(desc)), true, charset == null  ?  consoleDefaultCharset : charset);
```
assuming `ConsolePrintStream.localize(out, autoFlush, consoleDefaultCharset)` is equivalent to `ConsolePrintStream.localize(out, autoFlush)`.

Reply:
> assuming ConsolePrintStream.localize(out, autoFlush, consoleDefaultCharset) is equivalent to ConsolePrintStream.localize(out, autoFlush)

It is not. This change could be done for PrintStream, but it seems more confusing to do it this way.